react-app/src/components/Header/index.js

```diff
@@ -1,40 +1,48 @@
-import React, { PureComponent } from 'react';
+import React from 'react';
 import PropTypes from 'prop-types';
 
 import Logo from 'components/Logo';
 import styles from './index.module.scss';
 
-export default class Header extends PureComponent {
-  static propTypes = {
-    /**
-     * The element (an ActionIcon) that this element will show in the left.
-     */
-    left: PropTypes.element,
-
-    /**
-     * The element (an ActionIcon) that this element will show in the right.
-     */
-    right: PropTypes.element,
-  };
-
-  static defaultProps = {
-    left: null,
-    right: null,
-  }
-
-  render() {
-    return (
-      <div className={styles.container}>
-        <div className={styles.left}>
-          {this.props.left}
-        </div>
-        <div className={styles.center}>
-          <Logo size={Logo.SMALL} />
-        </div>
-        <div className={styles.right}>
-          {this.props.right}
-        </div>
+const Header = ({ left, right, whiteLogo }) => {
+  const logo = Logo({ white: whiteLogo });
```

Comment:
You shouldn't use this syntax, it relies in the component being a function.
Unless there's a good reason to do so, you should not use your component like this.

If you migrate your `Logo` component from a function to a class this will break.

You should use this syntax:
```jsx
const logo = <Logo white={whiteLogo} />;
```

Which works for components no matter how they are implemented.
I've seen this in several places, this comment applies to all use cases.

Reply:
Sadly, I lost again against premature optimisation. After reading [this article](https://medium.com/missive-app/45-faster-react-functional-components-now-3509a668e69f), I thought that it could be a good way to improve the performance of the app, but your point it's correct.

Reply:
Oh, nice! I remember that article. That came to my mind when I saw the syntax.
But yes, I would go that way unless we measure and there's a noticeable difference.
Also, things have changed a lot since the article was written.